From 339e32f8da17a956798ac110de2447abc64db1f8 Mon Sep 17 00:00:00 2001
From: jpadmana <jayashree.r.padmanaban@intel.com>
Date: Wed, 10 Apr 2013 12:39:58 +0530
Subject: [PORT FROM MAIN]MP3 album playback: extra power vs. single song playback -> mp3 decoder alternatively executed on IA and LPE?

BZ: 90129

The check for the flag mMusicOffloadOutput in isOffloadSupported
function should be used in combination with sessionId for gapless
offload scenarios.

Change-Id: I7df6d202b08f68505283ae9ab179603595095314
Orig-Change-Id: I113a38b8d22a5e27846f56fdf426f2cda6e63a77
Signed-off-by: jpadmana <jayashree.r.padmanaban@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 include/media/AudioSystem.h                  |    1 +
 include/media/IAudioPolicyService.h          |    1 +
 media/libmedia/AudioSystem.cpp               |    3 ++-
 media/libmedia/IAudioPolicyService.cpp       |    5 ++++-
 media/libstagefright/AwesomePlayer.cpp       |    1 +
 services/audioflinger/AudioPolicyService.cpp |    2 ++
 services/audioflinger/AudioPolicyService.h   |    1 +
 7 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/include/media/AudioSystem.h b/include/media/AudioSystem.h
index 8de7964..a4fc282 100644
--- a/include/media/AudioSystem.h
+++ b/include/media/AudioSystem.h
@@ -254,6 +254,7 @@ public:
                                     uint32_t samplingRate,
                                     uint32_t bitRate,
                                     int64_t duration,
+                                    int sessionId,
                                     bool hasVideo = false,
                                     bool hasStreaming = false);
 
diff --git a/include/media/IAudioPolicyService.h b/include/media/IAudioPolicyService.h
index 2bc3013..e92e73c 100644
--- a/include/media/IAudioPolicyService.h
+++ b/include/media/IAudioPolicyService.h
@@ -101,6 +101,7 @@ public:
                                     uint32_t samplingRate,
                                     uint32_t bitRate,
                                     int64_t duration,
+                                    int sessionId,
                                     bool hasVideo = false,
                                     bool hasStreaming = false) const = 0;
 };
diff --git a/media/libmedia/AudioSystem.cpp b/media/libmedia/AudioSystem.cpp
index d19d113..9599adb 100644
--- a/media/libmedia/AudioSystem.cpp
+++ b/media/libmedia/AudioSystem.cpp
@@ -792,6 +792,7 @@ bool AudioSystem::isOffloadSupported(uint32_t format,
                                     uint32_t samplingRate,
                                     uint32_t bitRate,
                                     int64_t duration,
+                                    int sessionId,
                                     bool isVideo,
                                     bool isStreaming)
 {
@@ -804,7 +805,7 @@ bool AudioSystem::isOffloadSupported(uint32_t format,
     }
 
     return aps->isOffloadSupported(format, stream, samplingRate, bitRate,
-                duration, isVideo, isStreaming);
+                duration, sessionId, isVideo, isStreaming);
 #else
     ALOGI("isOffloadSupported is not supported");
     return false;
diff --git a/media/libmedia/IAudioPolicyService.cpp b/media/libmedia/IAudioPolicyService.cpp
index a21da09..917b59d 100644
--- a/media/libmedia/IAudioPolicyService.cpp
+++ b/media/libmedia/IAudioPolicyService.cpp
@@ -381,6 +381,7 @@ public:
                                     uint32_t samplingRate,
                                     uint32_t bitRate,
                                     int64_t duration,
+                                    int sessionId,
                                     bool isVideo,
                                     bool isStreaming) const
     {
@@ -392,6 +393,7 @@ public:
         data.writeInt32(static_cast <uint32_t>(samplingRate));
         data.writeInt32(static_cast <uint32_t>(bitRate));
         data.writeInt64(static_cast <int64_t>(duration));
+        data.writeInt32(sessionId);
         data.writeInt32(isVideo);
         data.writeInt32(isStreaming);
         remote()->transact(IS_OFFLOAD_SUPPORTED, data, &reply);
@@ -688,10 +690,11 @@ status_t BnAudioPolicyService::onTransact(
             uint32_t samplingRate = data.readInt32();
            uint32_t bitRate = data.readInt32();
             int64_t duration = data.readInt64();
+            int sessionId = data.readInt32();
             bool isVideo = data.readInt32();
             bool isStreaming = data.readInt32();
             bool isSupported = isOffloadSupported(format, stream, samplingRate,
-                                   bitRate, duration, isVideo, isStreaming);
+                                   bitRate, duration, sessionId, isVideo, isStreaming);
             reply->writeInt32(isSupported);
 #endif
             return NO_ERROR;
diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 0c6f1c2..444b4e1 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1705,6 +1705,7 @@ status_t AwesomePlayer::initAudioDecoder() {
                 sampleRate,
                 avgBitRate,
                 mDurationUs,
+                mAudioSink->getSessionId(),
                 (mVideoTrack != NULL && mVideoSource != NULL),
                 isStreamingHTTP()) && !(isAudioEffectEnabled())) )
     {
diff --git a/services/audioflinger/AudioPolicyService.cpp b/services/audioflinger/AudioPolicyService.cpp
index d6b415d..45b7ae0 100644
--- a/services/audioflinger/AudioPolicyService.cpp
+++ b/services/audioflinger/AudioPolicyService.cpp
@@ -1063,6 +1063,7 @@ bool AudioPolicyService::isOffloadSupported(uint32_t format,
                                     uint32_t samplingRate,
                                     uint32_t bitRate,
                                     int64_t duration,
+                                    int sessionId,
                                     bool isVideo,
                                     bool isStreaming) const
 {
@@ -1077,6 +1078,7 @@ bool AudioPolicyService::isOffloadSupported(uint32_t format,
                                                samplingRate,
                                                bitRate,
                                                duration,
+                                               sessionId,
                                                isVideo,
                                                isStreaming);
 #endif
diff --git a/services/audioflinger/AudioPolicyService.h b/services/audioflinger/AudioPolicyService.h
index be3f326..d7ba954 100644
--- a/services/audioflinger/AudioPolicyService.h
+++ b/services/audioflinger/AudioPolicyService.h
@@ -140,6 +140,7 @@ public:
                                     uint32_t samplingRate,
                                     uint32_t bitRate,
                                     int64_t duration,
+                                    int sessionId,
                                     bool isVideo = false,
                                     bool isStreaming = false) const;
 
-- 
1.7.4.1

